framework/aster-frame/src/arch/x86/timer/pit.rs

```diff
@@ -1,21 +1,189 @@
 // SPDX-License-Identifier: MPL-2.0
 
-//! used for PIT Timer
+//! The Programmable Interval Timer (PIT) chip (Intel 8253/8254) basically consists of an oscillator,
+//! a prescaler and 3 independent frequency dividers. Each frequency divider has an output, which is
+//! used to allow the timer to control external circuitry (for example, IRQ 0).
+//!
+//! Reference: https://wiki.osdev.org/Programmable_Interval_Timer
+//!
 
-use crate::config::TIMER_FREQ;
+use crate::{
+    arch::{
+        kernel::IO_APIC,
+        timer::TIMER_FREQ,
+        x86::device::io_port::{IoPort, WriteOnlyAccess},
+    },
+    trap::IrqLine,
+};
 
-use crate::arch::x86::device::io_port::{IoPort, WriteOnlyAccess};
+/// PIT Operating Mode.
+///
+/// Usually, only the rate generator, which is used to determine the base frequency of other timers
+/// (e.g. APIC Timer), and the Square wave generator, which is used to generate interrupts directly, are used.
+///
+/// Note that if IOAPIC is used to manage interrupts and square wave mode is enabled, the frequency at which
+/// clock interrupts are generated is `Frequency/2`.
+#[repr(u8)]
+pub enum OperatingMode {
+    /// Triggers an interrupt (only on channel 0) when the counter is terminated (1 -> 0).
+    /// The data port needs to be reset before the next interrupt.
+    /// ```text,ignore
+    ///            software reload counter
+    ///                      ⬇
+    ///               +------+             +----
+    ///               |      |             |
+    /// --------------+      +-------------+
+    /// ⬆            ⬆                   ⬆
+    /// init()   counter 1 -> 0         counter 1 -> 0
```

Comment:
Did you draw the diagram yourself? They look amazing.

Reply:
Yes, I think that it is difficult to see the timer information directly with text description, so I have written these diagrams according to some other timer specifications I have seen before.